Add Kubernetes view to Fleet Automation docs (#37510)
DataDog/documentation: content/en/agent/fleet_automation/fleet_view.md

{{< img src="agent/fleet_automation/fleet_automation_remote_flare.png" alt="The support tab for an Agent with the Send Flare button." style="width:100%;" >}}

## Kubernetes view

> You can also view Datadog Agents and OTel Collectors running in Kubernetes environments, giving you a unified view of your fleet across both host-based and containerized infrastructure.

Reviewer: ```suggestion
The Kubernetes view lets you see Datadog Agents and OTel Collectors running in Kubernetes environments, giving you a unified view of your fleet across both host-based and containerized infrastructure.
```

Reviewer: "Also" implies a contrast with something already established, but the section might be retrieved in isolation by AI agents, or directly navigated to by users.

> By default, Fleet View lists infrastructure as individual hosts. Use the {{< ui >}}View by infra type{{< /ui >}} toggle to switch to [Kubernetes view][4], which shows Agents by Kubernetes cluster instead.

Reviewer: ```suggestion
By default, Fleet View lists infrastructure as individual hosts. Use the {{< ui >}}View by infra type{{< /ui >}} toggle to switch to the [Kubernetes view][4], which shows Agents by Kubernetes cluster instead.
```

Reviewer: Suggesting to add `the` here for consistency with other view modes, such as `topology view` here:
https://github.com/DataDog/documentation/blob/38291c32b696e575bf077fa58d04f8e728cf9e7a/content/en/agent/fleet_automation/fleet_view.md?plain=1#L83

It looks like `Fleet View` is the product name and correctly has no article. The others are descriptive names for UI modes.

Author: That works for me!

Each row is a cluster managed by the [Datadog Operator][5] or Helm chart, and the Agents in it (Node Agents, Cluster Agent, Cluster Check Runners) appear grouped per cluster instead of as individual hosts.

### Prerequisites for Kubernetes view

> - [Remote Configuration][6] must be enabled for your organization

Reviewer 2: ⚠️ It's only required if you need to perform edits on the configuration with Fleet Automation. Having RC disabled does not prevent you from using most of the features of the Kubernetes view

Author: Ahh thank you. I will work this in

Some capabilities require specific versions:

> - To view the `DatadogAgent` custom resource configuration: Datadog Operator v1.24 or later

Reviewer 2: You can also view the configuration with the Datadog Helm Chart v3.157.0 or later. 
Note that you do not see the `DatadogAgent` custom resource but directly the Helm Chart values (`values.yaml`) in this case.

- To edit the `DatadogAgent` configuration from Fleet View: Datadog Operator v1.27 or later

> - To view the list of integrations running on a Cluster Agent: Agent v7.72.0 or later. On Agent v7.79.0 or later, the integration list also includes status.

Reviewer: ```suggestion
- To view the list of integrations running on a Cluster Agent: Agent v7.72.0 or later
- To include integration status in the integration list: Agent v7.79.0 or later
```

Reviewer: Minor suggestion to split this for easier scanning — this also allows the punctuation to be removed, making these items more consistent with the other unordered list items in this section.

Additionally, it looks like the unordered list items in the "Limitations" section _are_ punctuated. Could you confirm if the different punctuation styles are intentional here?

Author: Re: punctuation, just an oversight on my part! ~~I will punctuate these.~~ Actually I reworked this section and these are now fragments, so no periods.

Agreed on splitting it, but I will reword slightly.

### View Kubernetes clusters

Clusters are listed alphabetically by cluster name. The table lists each cluster's name, deployment method (Datadog Operator or Helm), namespace, Agent version, Agent pod status, readiness, age, and number of restarts.

To find a specific cluster, you can [search](#search-and-filter) by cluster name.

Click a cluster to view:

- Cluster details, such as environment and tags
- Cluster-level Agents (Cluster Agent and Cluster Check Runners)
- Node Agents

> For clusters managed with the Datadog Operator v1.24 or later, the {{< ui >}}Configuration{{< /ui >}} tab shows the cluster's configuration data. With Datadog Operator v1.27 or later, you can also edit the configuration from this tab.

Reviewer 2: A bit misleading since, as mentioned in my previous comment, using the Datadog Helm Chart allows you to see the Configuration through the Helm Chart values on the `Configuration` tab.

### Limitations

> Compared to host view, Kubernetes view has the following limitations:

Reviewer: ```suggestion
Compared to the default view, the Kubernetes view has the following limitations:
```

Reviewer: It looks like `host view` isn't mentioned elsewhere in the page

> - You cannot send remote support flares. [under development v7.80]

Reviewer: It looks like `[under development v7.80]` might have been left here unintentionally. Could you please confirm?

Reviewer 3: No, it's a feature that's going to be available for Agent version >= 7.80

Author: I did mean to incorporate this into the draft properly, this was just a note to self. Thank you! (Actually will probably leave it out until it's released)

> - You can see which OTel Collectors are running, but you cannot view their configuration in Kubernetes view.

Reviewer: ```suggestion
- You can see which OTel Collectors are running, but you cannot view their configuration in the Kubernetes view.
```

{{< img src="agent/fleet_automation/fleet_automation_remote_flare.png" alt="The support tab for an Agent with the Send Flare button." style="width:100%;" >}}

## Kubernetes view

> The Kubernetes view lets you see Datadog Agents and OTel Collectors running in Kubernetes environments, giving you a unified view of your fleet across both host-based and containerized infrastructure.

Reviewer: ```suggestion
The Kubernetes view lets you see Datadog Agents and OTel Collectors running in Kubernetes environments. It provides a unified view of your fleet across both host-based and containerized infrastructure.
```

Reviewer: I don't actually see any problem with this, but the pr-review skill has flagged it as an AI writing marker (editorial reference: "Afterthought participial phrases at sentence ends")

Author: Huh, interesting. I don't feel strongly about the original phrasing, so I'll take the suggestion!

By default, Fleet View lists infrastructure as individual hosts. Use the {{< ui >}}View by infra type{{< /ui >}} toggle to switch to the [Kubernetes view][4], which shows Agents by Kubernetes cluster instead.

> Each row is a cluster managed by the [Datadog Operator][5] or Helm chart, and the Agents in it (Node Agents, Cluster Agent, Cluster Check Runners) appear grouped per cluster instead of as individual hosts.

Reviewer: ```suggestion
Each row is a cluster managed by the [Datadog Operator][5] or Helm chart. The cluster's Node Agents, Cluster Agent, and Cluster Check Runners appear grouped together rather than as individual hosts.
```

Reviewer: Minor suggestion to reduce sentence length — also, the "per cluster" detail is already stated on line 112.

Author: I like it, thanks!

### Prerequisites for Kubernetes view

> Most of the Kubernetes view is available without any prerequisite version. Specific capabilities have the following requirements:

Author: Reworked this section. It's getting a little large/funky; open to better ideas about how to organize it, but I think this works for now

Reviewer: One possibility might be to create a table. I gave it a go and this is how it looked in a local build:
<img width="765" height="470" alt="kubernetes-view-table" src="https://github.com/user-attachments/assets/b8223f32-b3fa-4294-bcc0-1c9fcc41fa19" />

This collapses the nesting into one scannable table, moves the limitation sentence out of a bullet and into a note below, and keeps the flag names visible without the sub-bullet layer.

Reviewer 4: I second the idea of a table! Makes it much easier to parse/update

Author: Sweet, thank you @buraizu

Author: Changed to the table. Thank you again, I like this much better

- Integrations list: Agent v7.72.0 or later
- Integration status in the list: Agent v7.79.0 or later

### View Kubernetes clusters

Clusters are listed alphabetically by cluster name. The table lists each cluster's name, deployment method (Datadog Operator or Helm), namespace, Agent version, Agent pod status, readiness, age, and number of restarts.

To find a specific cluster, you can [search](#search-and-filter) by cluster name.

Click a cluster to view:

- Cluster details, such as environment and tags
- Cluster-level Agents (Cluster Agent and Cluster Check Runners)
- Node Agents

On the {{< ui >}}Configuration{{< /ui >}} tab, you can view configuration:

- **Datadog Operator v1.24 or later**: View the `DatadogAgent` custom resource configuration. With Datadog Operator v1.27 or later, you can also edit the configuration from this tab.
- **Datadog Helm Chart v3.157.0 or later**: View the Helm Chart values (`values.yaml`).

### Limitations

Compared to the default view, the Kubernetes view has the following limitations:

- You cannot send remote support flares.
- You can see which OTel Collectors are running, but you cannot view their configuration in the Kubernetes view.

> - Fleet Automation API access is not available.

Reviewer: ```suggestion
- You cannot access the Fleet Automation API.
```

Reviewer: Minor suggestion for consistency

Reviewer 4: Hmm this sounds to me more like the user doesn't have perms to use the Fleet Automation API. How about "Fleet Automation API access is not available"

Author: Agreed; I like the original phrasing better, so I will dismiss this suggestion
